website/content/en/docs/scorecard/custom-tests.md

```diff
@@ -0,0 +1,190 @@
+│       │       └── main.go
+│       └── custom-scorecard-tests
+└── internal
+    └── tests
+        └── tests.go
+```
+
+1. `bundle/` - Contains the bundle manifests and metadata under test.
+2. `bundle/tests/scorecard/config.yaml` - Configuration yaml to define and run scorecard tests.
+3. `images/custom-scorecard-tests/cmd/test/main.go` - Scorecard test binary.
+4. `internal/tests/tests.go` -  Contains the implementation of custom tests specific to the operator.
+
+#### Writing custom test logic:
+
+Scorecard currently implements a few [basic][basic_tests] and [olm][olm_tests] tests for the image bundle, custom resources and custom resource definitions. Additional tests specific to the operator can also be included in the test suite of scorecard.
+
+The `tests.go` file is where the custom tests are implemented in the sample test image project. These tests use `scapiv1alpha2.ScorecardTestResult` struct to populate the result, which is then converted to json format for the output. For example, the format of a simple custom sample test can be as follows:
+
+```Go
+package tests
+
+import (
+	"github.com/operator-framework/operator-registry/pkg/registry"
+	scapiv1alpha2 "github.com/operator-framework/operator-sdk/pkg/apis/scorecard/v1alpha2"
+)
+
+const (
+	CustomTest1Name = "customtest1"
+)
+
+// CustomTest1 
+func CustomTest1(bundle registry.Bundle) scapiv1alpha2.ScorecardTestResult {
+	r := scapiv1alpha2.ScorecardTestResult{}
+	r.Name = CustomTest1Name
+	r.Description = "Custom Test 1"
+	r.State = scapiv1alpha2.PassState
+	r.Errors = make([]string, 0)
+	r.Suggestions = make([]string, 0)
+
+  // Implement relevant custom test logic here
+
+	return r
+}
+```
+
+### Scorecard Configuration file:
+
+The [configuration file][config_yaml] includes the test definition and metadata to run the test. For the example `CustomTest1` function, the following fields should be specified in `config.yaml`.
+
+```yaml
+tests:
+- name: "customtest1"
+  image: quay.io/jemccorm/custom-scorecard-tests:dev
+  entrypoint: 
+  - custom-scorecard-tests
+  - customtest1
+  labels:
+    suite: custom
+    test: customtest1
+  description: an ISV custom test
+  ```
+
+The important fields to note here are:
+1. `image` - name and tag of the test image which was specified in the Makefile.
+2. `labels` - the name of the `test` and `suite` the test function belongs to. This can be specified in the `operator-sdk alpha scorecard` command to run the desired test.
+
+**Note**: `config.yaml` file need not necessarily be present in the same repository. The `operator-sdk alpha scorecard` command includes the `--config` flag using which the path to the config file can be provided.
+
+### Scorecard binary:
+
+The scorecard test image implementation requires the bundle under test to be present in the test image. The `GetBundle()` function reads the pod's bundle to fetch the manifests and scorecard configuration from the desired path.
+
+```Go
+	cfg, err := GetBundle("/bundle")
```

Comment:
The `GetBundle()` needs to be taken from SDK. Will need to modify in the sample test image.